examples/client.c

```diff
@@ -756,9 +759,9 @@ usage( const char *program, const char *version) {
   fprintf( stderr,
      "PSK Options (if supported by underlying (D)TLS library)\n"
      "\t-h match_hint_file\n"
-     "\t       \t\tThis is a file that contains one or more lines of Identity\n"
-     "\t       \t\tHints to match for new user and new pre-shared key\n"
-     "\t       \t\t(PSK) (comma separated) to be used. E.g., per line\n"
+     "\t       \t\tThis is a file that contains one or more lines of\n"
```

Comment:
According [RFC4279, ClientKeyExchange](https://tools.ietf.org/html/rfc4279#page-5) the client didn't send the `hint`, it just receives that from the server. So, how is that `hint` intended to be used on the client side? Should the client select the `identity` based on that?

Reply:
The server provided Identity Hint could be indicating different, say, services (which have different PSKs) and so this gives the client the ability to chose a new key and new user (identity) to continue the (D)TLS set up.

However, TLS1.3 does not support this as the client includes the Identity in the Client Hello before the server can send an Identity Hint.

If the parameter validate_ih_call_back is set in coap_dtls_cpsk_t, then libcoap downgrades to (D)TLS1.2 (with a warning) so that Identity Hints are supported.

Reply:
Sure, for the server.

But this is the help for the client. so, what is the **client** intended to do with a hint, provided at its command line?

Reply:
The server issues a hint, the client receives the hint and the client (D)TLS layer can optionally call a callback with the hint and the callback then returns a key and identity to use for the ongoing (D)TLS session setup.  See `SSL_set_psk_client_callback()`.

So, if the client invokes with the -h option, the user of the client can define alternative key/identity depending on the hint received from the server.

Reply:
If the -h option is not used, then the -k and -u options are used for the key and identity.

Reply:
I'm not sure, if that works. Using only the `hint` as key requires that multiple server's are using a system of hints or shared crendtials. Do you have feedback from users about that? (I assumed, that this may be used to inform an user, before entering the secret, or after the handshake has failed. But that may be wrong.)

If it's used for that, why does the documentation mention "new user"?
Even, if the user is known, when the server sends it's hint, then the credentials are selected by the hint, or?

Reply:
https://tools.ietf.org/html/rfc4279#section-2 also states
````
   Both clients and servers may have pre-shared keys with several
   different parties.  The client indicates which key to use by
   including a "PSK identity" in the ClientKeyExchange message (note
   that unlike in [SHAREDKEYS], the session_id field in ClientHello
   message keeps its usual meaning).  To help the client in selecting
   which identity to use, the server can provide a "PSK identity hint"
   in the ServerKeyExchange message.  If no hint is provided, the
   ServerKeyExchange message is omitted.  See Section 5 for a more
   detailed description of these fields.
````
So a client can choose an identity and possibly an associated pre-shared key based on receiving the hint.  I agree that `new` is confusing and perhaps `alternative` should be used.  The server will still choose the pre-shared key to use based on the received identity(user).

No specific feedback from the users -but it is functionality in TLS1.2 or earlier.

Reply:
Documentation updated to not use new and additional clarification on the server side.  Code pushed for review.